apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -73,11 +76,30 @@ const trackBlockReplacement = ( originalBlockIds, blocks ) => {
 	} );
 };
 
+/**
+ * Track inner blocks replacement.
+ * This is how Page Templates insert their content into page, by replacing everything that was already there.
+ *
+ * @param {Array} rootClientId id of parent block
+ * @param {object|Array} blocks block instance object or an array of such objects
+ * @returns {void}
+ */
+const trackInnerBlocksReplacement = ( rootClientId, blocks ) => {
+	castArray( blocks ).forEach( block => {
```

Comment:
If before `castArray()` `blocks` is a deeply nested object then presumably we can recurse into it and call the same callback you provide to `forEach` to handle dispatching events for _all_ blocks and not just the top level?

Reply:
Just checked and `blocks` is

> Block objects to insert as new InnerBlocks

To presumably it is nested.

Reply:
We can flatten the blocks here as you suggest. I haven't done it immediately as I was wondering if other actions we are tracking should also be looked at recursively, hence the suggestion to make it a separate task and get answers first.

Reply:
>  I was wondering if other actions we are tracking should also be looked at recursively

Yes, it's a very good point, too.